ols/app/models/config.py

```diff
@@ -120,6 +120,44 @@ def validate_yaml(self, disable_tls: bool = False) -> None:
             checks.file_check(self.tls_key_path, "OLS server certificate private key")
 
 
+class ProxyConfig(BaseModel):
+    """HTTPS Proxy configuration."""
+
+    proxy_url: Optional[AnyHttpUrl] = None
+    proxy_ca_cert_path: Optional[FilePath] = None
+
+    def __init__(self, data: Optional[dict] = None) -> None:
+        """Initialize configuration and perform basic validation."""
+        super().__init__()
+        if not data:
+            return
+        self.proxy_url = data.get("proxy_url")  # type: ignore
```

Comment:
Is `# type: ignore` really needed?

Reply:
yes. elsewise the type check will not pass 
```
pdm run mypy --explicit-package-bases --disallow-untyped-calls --disallow-untyped-defs --disallow-incomplete-defs ols/
INFO: Inside an active virtualenv /opt/app-root, reusing it.
Set env var PDM_IGNORE_ACTIVE_VENV to ignore it.
/opt/app-root/lib64/python3.11/site-packages/requests/__init__.py:86: RequestsDependencyWarning: Unable to find acceptable character detection dependency (chardet or charset_normalizer).
  warnings.warn(
ols/app/models/config.py:127: error: Incompatible types in assignment (expression has type "str | None", variable has type "AnyHttpUrl | None")  [assignment]
```